qiskit/opflow/gradients/natural_gradient.py

```diff
@@ -120,33 +125,65 @@ def convert(
         # Instantiate the QFI metric which is used to re-scale the gradient
         metric = self._qfi_method.convert(operator[-1], params) * 0.25
 
-        # Define the function which compute the natural gradient from the gradient and the QFI.
         def combo_fn(x):
-            c = np.real(x[0])
-            a = np.real(x[1])
-            if self.regularization:
-                # If a regularization method is chosen then use a regularized solver to
-                # construct the natural gradient.
-                nat_grad = NaturalGradient._regularized_sle_solver(
-                    a, c, regularization=self.regularization
-                )
-            else:
-                try:
-                    # Try to solve the system of linear equations Ax = C.
-                    nat_grad = np.linalg.solve(a, c)
-                except np.linalg.LinAlgError:  # singular matrix
-                    nat_grad = np.linalg.lstsq(a, c)[0]
-            return np.real(nat_grad)
+            return self.nat_grad_combo_fn(x, self.regularization)
 
         # Define the ListOp which combines the gradient and the QFI according to the combination
         # function defined above.
         return ListOp([grad, metric], combo_fn=combo_fn)
 
+    @staticmethod
+    def nat_grad_combo_fn(x: tuple, regularization: Optional[str] = None) -> np.ndarray:
+        r"""
+        Natural Gradient Function Implementation.
+
+        Args:
+            x: Iterable consisting of Gradient, Quantum Geometric Tensor.
+            regularization: Regularization method.
+
+        Returns:
+            Natural Gradient.
+
+        Raises:
+            ValueError: If the gradient has imaginary components that are non-negligible.
+
+        """
+        gradient = x[0]
+        metric = x[1]
+        if any(np.abs(np.imag(c_item)) > ETOL for c_item in gradient):
+            raise ValueError("The imaginary part of the gradient are non-negligible.")
+        gradient = np.real(gradient)
+
+        if regularization is not None:
+            # If a regularization method is chosen then use a regularized solver to
+            # construct the natural gradient.
+            nat_grad = NaturalGradient._regularized_sle_solver(
+                metric, gradient, regularization=regularization
+            )
+        else:
+            # Check if numerical instabilities lead to a metric which is not positive semidefinite
+            w, v = np.linalg.eigh(metric)
+
+            if not all(ew >= (-1) * ETOL for ew in w):
+                raise ValueError(
+                    "The underlying metric has ein Eigenvalue < -1e-8. "
+                    "Please use a regularized least-square solver for this problem or"
+                    "increase the number of backend shots.",
```

Comment:
Similar comment about including formatted information to show the user the magnitude of the error